bin/blocks-plot

```diff
@@ -0,0 +1,256 @@
+        Which channels 
+
+    epochs_only: bool
+        If True, all; log-records which do not correspond to a full epoch w
+        be filtered out. Full epochs are identified by having a 'epoch-took'
+        record, or by having a time of 0.
+
+    Returns
+    -------
+    DataFrame
+    """
+    channels = defaultdict(dict)
+    for t, channel, value in log:
+        # TODO: ensure value is something we actually want to keep.
+        # We probably have to discard all non-scalars as they can not
+        # be stored in a DataFrame.
+        # TODO: ensure that the channels have "homogeneous" data types
+        channels[channel] [t] = value
+
+    df = pandas.DataFrame(channels)
+
+    if epochs_only:
+        if 'epoch_took' not in df.columns:
+            raise ValueError("Can't filter for epochs: "\
+                        "Log does not contain 'epoch_took'")
+        mask = pandas.notnull(df.epoch_took) | (df.index == 0)
+        df = df[mask]
+
+    return df
+
+def match_channel_specs(experiments, channel_specs):
+    """
+    Parameters
+    ----------
+    experiments : OrderedDict of (str: DataFrames)
+    channel_specs : list of str
+
+    Returns
+    -------
+    a DataFrame with the specified channels`
+    """
+    df = pandas.DataFrame()
+
+    # This is not pretty...
+    #  We iterate over all channel_specs and match each spec to the
+    #  channels of all experiments.
+    for spec in channel_specs:
+        if ":" in spec:
+            exp_spec, channel_spec = spec.split(":")
+        else:
+            exp_spec, channel_spec = None, spec
+
+        for i, exp in enumerate(experiments.values()):
+            for channel in fnmatch.filter(exp.columns, channel_spec):
+                if exp_spec and exp_spec != i:
+                    # We are looking for a specific experiment..
+                    #  ... and it's not this one.
+                    continue
+
+                column_name = "{}:{}".format(i, channel)
+                df[column_name] = exp[channel]
+
+    return df
+
+
+def plot_dataframe(dataframe):
+    import pylab
+    import matplotlib as mpl
+
+    t = dataframe.index
+    print("Plotting {} channels:".format(len(dataframe.columns)))
+    for cname, series in iteritems(dataframe):
+        print("    {}".format(cname))
+        pylab.plot(t, series, label=cname)
+    pylab.legend()
+    pylab.show(block=True)
+
+PROMPT_HEADER = """
+
+Type a comma separated list of channels to plot or [q]uit.
+
+Channels may be prefixed by <number>: to refer to a specific experiment and may
+contain '*' or '?' characters to match multiple channels at once.
+
+"""
+
+IPYTHON_HEADER = """
+
+Complete DataFrames for all experiments can be found in the 'experiments[fname]'
+dictionary. The DataFrame containing only the selected channels can be access
+as 'matched'
+
+"""
+
+def main(args):
+    # Load and convert experiments into DataFrames...
+    experiments = OrderedDict()
+    for fname in args.experiments:
```

Comment:
That might be implemented in future revisions, but ideally I would like to be able to load pickles consisting of a log only as well, and maybe even dataframes.